import_export/widgets.py

```diff
@@ -267,6 +269,21 @@ def clean(self, value, row=None, *args, **kwargs):
     def render(self, value, obj=None):
         return self.separator.join(six.text_type(v) for v in value)
 
+    
+class JSONWidget(Widget):
+    """
+    Widget for a JSON object (especially required for jsonb fields in PostgreSQL database.)
+    """
+
+    def clean(self, value, row=None, *args, **kwargs):
+        val = super(JSONWidget, self).clean(value)
+        if val:
+            return ast.literal_eval(val)
```

Comment:
Why not json.loads here?

Reply:
I'm guessing it's because of stuff like this:

```python
>>> my_json = "{'single-quoted strings': 'aren\\\'t allowed by the JSON spec'}"
>>> json.loads(my_json)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/decoder.py", line 355, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)
>>> ast.literal_eval(my_json)
{'single-quoted strings': "aren't allowed by the JSON spec"}
```

The good news is that `ast.literal_eval()` won't evaluate expressions or statements, only Python literals, so it's pretty safe.

Reply:
I don’t understand the first line!
JSON requires double quotes, why pass something else?

Reply:
It is because on importing a file, a string is imported not JSON.

Reply:
I’m sorry I don’t get it.

A JSON file should have contents like `{"key": "value"}`.
Where does the example with single quotes come from?

Reply:
I agree to that. However this widget is not just for exporting/importing JSON files (to make it clear, it should handle other file formats too). It's mainly provides support PostgreSQL DB which supports JSON field and when you need to export/import them from any file formats. And by Postel's law, it is liberal in accepting any form of JSON from any file format and is also conservative as it only gives clear JSON as output for json field in PostgreSQL DB.

> I’m sorry I don’t get it.
> 
> A JSON file should have contents like `{"key": "value"}`.
> Where does the example with single quotes come from?

Reply:
Oops. I hereby apologise profusely for using the word _probably_ above rather than _definitely_, because `ast.literal_eval` can't handle `null`:

```python
>>> my_json = '{"ohdear": null}'
>>> json.loads(my_json)
{'ohdear': None}

>>> ast.literal_eval(my_json)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/yoz/anaconda3/envs/insights-peops/lib/python3.6/ast.py", line 85, in literal_eval
    return _convert(node_or_string)
  File "/Users/yoz/anaconda3/envs/insights-peops/lib/python3.6/ast.py", line 66, in _convert
    in zip(node.keys, node.values))
  File "/Users/yoz/anaconda3/envs/insights-peops/lib/python3.6/ast.py", line 65, in <genexpr>
    return dict((_convert(k), _convert(v)) for k, v
  File "/Users/yoz/anaconda3/envs/insights-peops/lib/python3.6/ast.py", line 84, in _convert
    raise ValueError('malformed node or string: ' + repr(node))
ValueError: malformed node or string: <_ast.Name object at 0x105e07e48>
```

Reply:
Thats true. Thanks.
[Pull request](https://github.com/django-import-export/django-import-export/pull/928)